### PR TITLE
Add support for Fireworks AI

### PR DESCRIPTION
**Pull request by dqbd**
Adds a new LLM class and Chat Model class.

### CODE REVIEW

File: langchain/src/chat_models/fireworks.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    delete request.functions;
+
+    if (request.stream === true) {
+      return super.completionWithRetry(request, options);
+    }
+
+    return super.completionWithRetry(request, options);
+  }
+
+  bind(
```
Comment (jacoblee93) [inline langchain/src/chat_models/fireworks.ts:124]: This is fine but should we just add a `CallOption` type arg to OpenAI/ChatOpenAI instead?

File: langchain/src/llms/tests/fireworks.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { test, expect } from "@jest/globals";
+import { Fireworks } from "../fireworks.js";
+
+describe("Vertex AI", () => {
```
Comment (jacoblee93) [inline langchain/src/llms/tests/fireworks.int.test.ts:4]: Change

File: langchain/src/chat_models/fireworks.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    | AsyncIterable<OpenAIClient.Chat.Completions.ChatCompletionChunk>
+    | OpenAIClient.Chat.Completions.ChatCompletion
+  > {
+    delete request.frequency_penalty;
+    delete request.presence_penalty;
+    delete request.logit_bias;
+    delete request.functions;
+
+    if (request.stream === true) {
+      return super.completionWithRetry(request, options);
```
Comment (jacoblee93) [inline langchain/src/chat_models/fireworks.ts:118]: Oh wait what's this?

Comment (dqbd) [reply langchain/src/chat_models/fireworks.ts:118]: TS has some weird type widening issues, so this is to nudge TS that we're using the correct method

File: langchain/src/chat_models/fireworks.ts (hunk 1 above)
Comment (dqbd) [reply langchain/src/chat_models/fireworks.ts:124]: That's a possibility, but I feel kind of weird modifying the OpenAI class to support other LLMs, wdyt?

**Pull request merged**